ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/resources/arjunacore/XAResourceRecord.java

```diff
@@ -227,10 +225,15 @@ public int topLevelPrepare()
 			case XAException.XA_RBOTHER:
 			case XAException.XA_RBPROTO:
 			case XAException.XA_RBTIMEOUT:
+				// This will avoid calling rollback when aborting
+				_rolledBack = true;
```

Comment:
I am wondering if these _rolledBack should be replaced with calls to removeConnection() or at least augmented with them. I am not super sure of it but it looks like removeConnection() is intended to disconnect the XAResource from the transaction which then is somehow conditionally used to avoid doing topLevelAbort. That being said topLevelAbort would still execute in some cases due to the conditionality and we don't want that I think from an XA point of view (and when it is suppressed, it would return FINISH_OK which I am not sure if it's always correct but that is how it is written today).

By only setting this new variable _rolledBack rather than calling removeConnection(), do you know if the _recoveryObject will be closed in the same way as it was for all cases?

Reply:
I might be running into a similar problem with a pull request I have: https://github.com/jbosstm/narayana/pull/2215/files#r1516032102 (in my case I am avoiding a call to destroyState() [which would call removeConnection()])

Reply:
It is possible I was running into this kind of problem, my change had previously prevented removeConnection from being executed. I have fixed it in a commit I have made here https://github.com/jbosstm/narayana/pull/2215/commits/a4f7f5153b65a72ba466c4e49af0d7032b296952

I might be squashing this before long though so I will include the point here.

I had something like:
```
					if (_prepared)
						if (destroyState)
							destroyState();
					else
						removeConnection();
```
And I think it should be:
```
					if (_prepared && destroyState)
						destroyState();
					else
						removeConnection();
```

Reply:
If you are already making changes related to this logic, probably doesn't even need to be a change to this particular statement but in the logic of how connections are handled, then I don't think our recommended policies would prohibit you from making that change, ie you would not need to wait for a reformat of the whole file. 

@tomjenkinson In other words, if your change clarifies the area that you are fixing then we are flexible enough to see it as of benefit and related to the fix and we would block the PR using arguments based on rigid application of our re-formatting rules.

Reply:
But to answer your question, I think the way the author probably intended to code the boolean condition the way you coded it. So I agree that it's a good change and an example of why the braces should be mandatory when coding up this kind of logic.